asdf/schema.py

```diff
@@ -569,11 +587,18 @@ def _iter_errors(self, instance, _schema=None):
 
                 for schema_uri in schema_uris:
                     try:
-                        tag_schema = self._resolver.resolve(schema_uri)[1]
-                        yield from self._json_schema_validator_factory.create(tag_schema, self._resolver).iter_errors(
-                            node,
-                        )
-                    except RefResolutionError:
+                        if USE_REFERENCING:
+                            tag_resource = self._resolver.get_or_retrieve(schema_uri)
+                            resolver = tag_resource.value @ self._resolver
+                            v = self._json_schema_validator_factory.create(tag_resource.value.contents, resolver)
+                            v._resolver = resolver.resolver_with_root(tag_resource.value)
```

Comment:
This was added when trying to debug why many tests were failing with errors like:
```
referencing.exceptions.Unresolvable: software-1.0.0
```
which was tracked down to loss of the `base_uri` I think here:
https://github.com/python-jsonschema/jsonschema/blob/529b57aefb4810866c3de0750349fb227fac816e/jsonschema/validators.py#L248-L252